r/R/dplyr-funcs-datetime.R

```diff
@@ -105,7 +105,22 @@ register_bindings_datetime <- function() {
     (call_binding("yday", x) - 1) %/% 7 + 1
   })
 
-  register_binding("month", function(x, label = FALSE, abbr = TRUE, locale = Sys.getlocale("LC_TIME")) {
+  register_binding("month", function(x,
+                                     label = FALSE,
+                                     abbr = TRUE,
+                                     locale = Sys.getlocale("LC_TIME")) {
+
+    if (call_binding("is.integer", x)) {
+      if (is.integer(x)) {
+        x <- build_expr("cast", x, options = cast_options(to_type = int32()))
+      }
+      x <- call_binding("if_else",
+                        call_binding("between", x, 1, 12),
+                        x,
+                        NA_integer_)
+      x <- build_expr("cast", x * 28L, options = cast_options(to_type = date32()))
```

Comment:
This is a cheeky implementation, but it works. 😄

Reply:
We could actually short-circuit here and return the expression with simply `if_else` when `label = FALSE`, since that will return an integer itself, yeah? 

I agree that this is a little bit unorthodox, but I _think_ it works. An alternative would be to put this value into a date string with a call to the `paste` binding and then doing `strptime` on that. Both aren't super ideal — but I suspect that this integer multiplication is going to be more performant (though if we are basing a decision on that, we probably should measure it!).

Reply:
I haven't done any benchmarking yet.